lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -524,15 +524,83 @@ sub add_comment {
     my $job = $self->app->schema->resultset("Jobs")->find($self->param('testid'));
     return $self->reply->not_found unless $job;
 
+    # only logged in users can add comments
+    if (!$self->current_user) {
+        $self->flash('info', 'The comment couldn\'t be added because you\'re not logged in anymore');
+        return $self->redirect_to('test');
+    }
```

Comment:
… ok, more like triple now. I understand that the redirect target differs but that could either be a helper function parameter or maybe better there is some attribute in self which refers to the source where you came from.